be/src/olap/rowset/segment_v2/binary_plain_page.h

```diff
@@ -107,19 +113,46 @@ class BinaryPlainPageBuilder : public PageBuilder {
         return _size_estimate;
     }
 
+    Status get_first_value(void* value) const override {
+        DCHECK(_finished);
+        if (_offsets.size() == 0) {
+            return Status::NotFound("page is empty");
+        }
+        *reinterpret_cast<Slice*>(value) = Slice(_first_value);
+        return Status::OK();
+    }
+    Status get_last_value(void* value) const override {
+        DCHECK(_finished);
+        if (_offsets.size() == 0) {
+            return Status::NotFound("page is empty");
+        }
+        *reinterpret_cast<Slice*>(value) = Slice(_last_value);
+        return Status::OK();
+    }
+
     void update_prepared_size(size_t added_size) {
         _prepared_size += added_size;
         _prepared_size += sizeof(uint32_t);
     }
 
 private:
+    void _copy_value_at(faststring& value, size_t idx) const {
```

Comment:
```suggestion
    void _copy_value_at(faststring* value, size_t idx) const {
```
In c++, we prefer using pointer param as output param, and const reference as input param

Reply:
oh. I know it but forgot.